toolchain/check/impl_lookup.cpp

```diff
@@ -246,6 +250,8 @@ static auto GetWitnessIdForImpl(Context& context, SemIR::LocId loc_id,
   // type: the `I` in `impl ... as I`. The deduction step may be unable to be
   // fully applied to the types in the constraint and result in an error here,
   // in which case it does not match the query.
+  // TODO: Should this be using `impl.interface` instead of
+  // `impl.constraint_id`?
```

Comment:
Hm, that would prevent it from seeing undeducable things in the constraint and making an error here. Do you have an example you're worried about here?